garage/tf/algos/reps.py

```diff
@@ -0,0 +1,442 @@
+                    action_flat,
+                    valid_flat,
+                    name="action_valid")    # yapf: disable
+                policy_state_info_vars_valid = filter_valids_dict(
+                    policy_state_info_vars_flat,
+                    valid_flat,
+                    name="policy_state_info_vars_valid")
+                policy_old_dist_info_vars_valid = filter_valids_dict(
+                    policy_old_dist_info_vars_flat,
+                    valid_flat,
+                    name="policy_old_dist_info_vars_valid")
+
+        pol_flat = graph_inputs(
+            "PolicyLossInputsFlat",
+            obs_var=obs_flat,
+            action_var=action_flat,
+            reward_var=reward_flat,
+            valid_var=valid_flat,
+            feat_diff=feat_diff_flat,
+            policy_state_info_vars=policy_state_info_vars_flat,
+            policy_old_dist_info_vars=policy_old_dist_info_vars_flat,
+        )
+        pol_valid = graph_inputs(
+            "PolicyLossInputsValid",
+            reward_var=reward_valid,
+            action_var=action_valid,
+            policy_state_info_vars=policy_state_info_vars_valid,
+            policy_old_dist_info_vars=policy_old_dist_info_vars_valid,
+        )
+        policy_loss_inputs = graph_inputs(
+            "PolicyLossInputs",
+            obs_var=obs_var,
+            action_var=action_var,
+            reward_var=reward_var,
+            valid_var=valid_var,
+            feat_diff=feat_diff,
+            param_eta=param_eta,
+            param_v=param_v,
+            policy_state_info_vars=policy_state_info_vars,
+            policy_old_dist_info_vars=policy_old_dist_info_vars,
+            flat=pol_flat,
+            valid=pol_valid,
+        )
+        policy_opt_inputs = graph_inputs(
+            "PolicyOptInputs",
+            obs_var=obs_var,
+            action_var=action_var,
+            reward_var=reward_var,
+            valid_var=valid_var,
+            feat_diff=feat_diff,
+            param_eta=param_eta,
+            param_v=param_v,
+            policy_state_info_vars_list=policy_state_info_vars_list,
+            policy_old_dist_info_vars_list=policy_old_dist_info_vars_list,
+        )
+        dual_opt_inputs = graph_inputs(
+            "DualOptInputs",
+            reward_var=reward_var,
+            valid_var=valid_var,
+            feat_diff=feat_diff,
+            param_eta=param_eta,
+            param_v=param_v,
+            policy_state_info_vars_list=policy_state_info_vars_list,
+            policy_old_dist_info_vars_list=policy_old_dist_info_vars_list,
+        )
+
+        return policy_loss_inputs, policy_opt_inputs, dual_opt_inputs
+
+    def _build_policy_loss(self, i):
+        """Initialize policy loss complie function based on inputs i."""
+        pol_dist = self.policy.distribution
+        is_recurrent = self.policy.recurrent
+
+        # Initialize dual params
+        self.param_eta = 15.
```

Comment:
Is there any particular reason why eta is 15?

Reply:
we can only squeeze so much magic out of hyperparameters, unfortunately.